hvp.php

```diff
@@ -248,21 +392,71 @@ public function saveLibraryDependencies($libraryId, $dependencies, $dependency_t
   }
 
   /**
-   * Implements saveContentData
+   * Implements updateContent
+   *
+   * Updates existing H5P content.
+   *
+   * @param array $content
+   *   An associative array containing:
+   *   - id: The content id
+   *   - params: The content in json format
+   *   - library: An associative array containing:
+   *     - libraryId: The id of the main library for this content
+   * @param int $contentMainId
+   *   Main id for the content if this is a system that supports versioning
    */
-  public function saveContentData($contentId, $contentJson, $mainJsonData, $mainLibraryId, $contentMainId = NULL) {
+  public function updateContent($content, $contentMainId = NULL) {
     global $DB;
-    
-    $DB->insert_record_raw('hvp_contents', array(
-      'id' => $contentId,
-      'content' => $contentJson,
-      'embed_type' => (isset($mainJsonData['embedTypes']) ? implode(', ', $mainJsonData['embedTypes']) : ''),
-      'library_id' => $mainLibraryId
-    ), false, false, true);
+
+    // There's always an id because the H5P entry is first saved to the
+    // 'hvp' table, and the same id is being used in the 'hvp_contents'
+    // table. Therefore we must check manually whether the row exists.
+    $existing = $DB->get_record('hvp_contents', array('id' => $content['id']));
```

Comment:
Yes, unfortunately, numbers aren't allowed in the module name/slug.
Regarding the tables, I think maybe the *hvp* and *hvp_contents* could be a single table, like in the WordPress plugin. I'm not sure why it isn't, it might be a Moodle thing. I'll look into it.